modules/protocols/snap/include/virgil/iot/protocols/snap/snap-structs.h

```diff
@@ -296,7 +312,7 @@ typedef struct __attribute__((__packed__)) {
  * This structure contains network interface callbacks and packet data
  */
 typedef struct vs_netif_t {
-    void *user_data; /**< User data */
+    vs_snap_service_user_data_t user_data; /**< User data */
```

Comment:
Why we use vs_snap_service_user_data_t into vs_netif_t struct instead of  vs_netif_user_data_t?